### ISSUE TITLE
Eager load on nested items only populates first nested item in the collection 

### ISSUE THREAD

**Issue by subsociety** [gobuffalo/pop#123]
I have the following model structure:

```
type BenefitCategory struct {
	ID              int64  `db:"id"`
	Name        string `db:"name"`

	Benefits []Benefit `many_to_many:"benefit_category_benefits"`
}

type Benefit struct {
	ID                      int64  `db:"id"`
	Name               string  `db:"name"`

	CancellationRuleID *int64 `db:"cancellation_rule_id"`
	CancellationRule CancellationRule `belongs_to:"cancellation_rule"`
}

type CancellationRule struct {
	ID               int64  `db:"id"`
	Name        string  `db:"name"`
}
```

When specifying eager loading on BenefitCategory for the nested Benefits.CancellationRule, I'm only seeing the nested values populated for the first BenefitCategory.  

```
categories = &[]BenefitCategory[]
err := tx.Eager("Benefits.CancellationRule").All(categories)
```

Execution of the call above results in the data:

```
{
    "benefit_categories": [
        {
            "id": 37,
            "name": "Category 1",
            "benefits": [
                {
                    "id": 79,
                    "name": "Benefit 1",
                    "cancellation_rule_id": 5,
                    "cancellation_rule": {
                        "id": 5,
                        "name": "Strict"
                    }
                },
                {
                    "id": 80,
                    "name": "Benefit 2",
                    "cancellation_rule_id": 5,
                    "cancellation_rule": {
                        "id": 5,
                        "name": "Strict"
                    }
                }
            ]
        },
        {
            "id": 38,
            "name": "Category 2",
            "benefits": [
                {
                    "id": 83,
                    "name": "Benefit 3",
                    "cancellation_rule_id": 5,
                    "cancellation_rule": {
                        "id": 0,
                        "name": "",
                        "description": ""
                    }
                },
                {
                    "id": 85,
                    "name": "Benefit 4",
                    "cancellation_rule_id": 5,
                    "cancellation_rule": {
                        "id": 0,
                        "name": "",
                        "description": ""
                    }
                }
            ]
        }, ...
    ]
}
```

The nested _cancellation_rule_ values are only populated for the first item in the top level collection of _benefit_categories_.  Debug statements in the _pop.eagerAssocations()_ method hint that subsequent items in the collection are not processed.

**Comment by robbyoconnor** [comment]
This is a limitation sadly...it will only populate one level.

**Comment by larrymjordan** [comment]
@subsociety can you give us output log thrown for this issue?

**Comment by mclark4386** [comment]
@robbyoconnor he is actually loading it correctly. Look at the pasted output. Specifically the cascellation_rules at the beginning versus the end. 

@subsociety like @larrymjordan said please add any log output as well.

**Comment by larrymjordan** [comment]
@subsociety I would like to see if any error was raised while loading inner level. However, I could replicate it and you're correct about this. It seems this happens when loading nested associations with the has_many type. I'm working on this fix.

**Comment by subsociety** [comment]
Thanks @larrymjordan, I can verify that there are no errors output in the logs.

**Comment by robbyoconnor** [comment]
@mclark4386 ah so he is!  I guess I didn't read it close enough -- sorry @subsociety !

**Comment by stanislas-m** [comment]
Fixed with #124.

**Comment by subsociety** [comment]
Awesome fix, thanks guys!
